### ISSUE TITLE
tf.pack() issue while building the model.. 

### ISSUE THREAD

**Issue by kannadaraj** [inikdom/neural-sentiment#16]
Hi Dominik,

Thanks very much for this great tool. I tried to use this tool kit and i am ending up with an error. I have all the dependencies. 
The code downloads the database, builds the vocab and 4 data files in processed directory and than when i am trying to build a model it give the following error. Please can you suggest what might be the problem.. 

Traceback (most recent call last):
  File "train.py", line 149, in <module>
    main()
  File "train.py", line 69, in main
    model = createModel(sess, vocab_size)
  File "train.py", line 123, in createModel
    FLAGS.learning_rate, FLAGS.lr_decay_factor)
  File "/home/neural-sentiment/models/sentiment.py", line 59, in **init**
    concat_states =  tf.pack(encoder_state)
  File "/home/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/ops/array_ops.py", line 228, in pack
    return gen_array_ops._pack(values, name=name)
  File "/home/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/ops/gen_array_ops.py", line 523, in _pack
    return _op_def_lib.apply_op("Pack", values=values, name=name)
  File "/home/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/ops/op_def_library.py", line 357, in apply_op
    (input_name, op_type_name, values))
TypeError: Expected list for 'values' argument to 'Pack' Op, not Tensor("RNN/concat_199:0", shape=(?, 200), dtype=float32).

regards
Kannadaraj

**Comment by domerin0** [comment]
Hello @kannadaraj ,

update:*\* I was able to recreate the issue in a clean vm install, just trying to figure out what is causing it.

Thanks for bringing this issue to my attention. I'm trying to recreate it on my end. So far, cloning master branch, and running `$ python train.py` seems to do what I expect. I was wondering if you are running with the default hyper parameters or using your own (if you are using your own, do mind letting me know which ones you changed to help me re-create the issue)?

I am asking because I've recently realized that running alternative hyper parameters causes exceptions similar to that. It doesn't check to see if the model is compatible. My plan is to do that check, then create a new model if it is not compatible. I will try to make that fix by tonight. So far this is my best guess as to what the problem is.

I'm going to admit I'm not confident that is the actual problem from reading your stack trace. I will also do a clean install of everything in a vm later today, just incase there is some sort of dependency issue I've missed. What OS are you running this on?

**Comment by domerin0** [comment]
Hello @kannadaraj 

I believe I've found the root of the problem. I built this with Tensorflow version 0.6.0, and the issue occurs in version 0.7.1. According to the release notes found [here](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md)

`The non-public nn.rnn and the various nn.seq2seq methods now return just the final state instead of the list of all states.`

Unfortunately this means that the states can't be averaged over time anymore. I have committed a 'fix'. Master should be good to pull and run now. I also partially fixed the inability to change hyper parameters. I'm not sure how this will affect network performance and training, I'll have to test it out over the next few days.
